Solutions/Marain.TenantManagement.Abstractions/Marain/TenantManagement/Configuration/CosmosConfigurationItem.cs

```diff
@@ -0,0 +1,45 @@
+﻿// <copyright file="CosmosConfigurationItem.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.TenantManagement.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using Corvus.Azure.Cosmos.Tenancy;
+
+    /// <summary>
+    /// Enrollment configuration item for tenanted cosmos storage config.
+    /// </summary>
+    public class CosmosConfigurationItem : StorageConfigurationItem<CosmosConfiguration>
+    {
+        /// <summary>
+        /// The content type of the configuration item.
+        /// </summary>
+        public const string RegisteredContentType = ConfigurationItem.BaseContentType + "cosmosdb";
+
+        /// <inheritdoc/>
+        public override string ContentType => RegisteredContentType;
+
+        /// <inheritdoc/>
+        public override IEnumerable<KeyValuePair<string, object>> AddConfiguration(IEnumerable<KeyValuePair<string, object>> values)
+        {
+            if (this.Configuration.DatabaseName == null)
+            {
+                throw new NullReferenceException($"{nameof(this.Configuration.DatabaseName)} cannot be null.");
+            }
+
+            if (this.Configuration.ContainerName == null)
+            {
+                throw new NullReferenceException($"{nameof(this.Configuration.DatabaseName)} cannot be null.");
+            }
```

Comment:
It's valid for either or both of `DatabaseName` and `ContainerName` to be null - we fall back to those provided in the `CosmosContainerDefinition` if they aren't provided. See https://github.com/corvus-dotnet/Corvus.Tenancy/blob/2c49f150c9fa093b7ac983175694f16d3436061a/Solutions/Corvus.Azure.Cosmos.Tenancy/Corvus/Azure/Cosmos/Tenancy/Internal/TenantCosmosContainerFactory.cs#L215

Reply:
Removed validation